pycbc/io/hdf.py

```diff
@@ -681,12 +681,27 @@ class ForegroundTriggers(object):
     def __init__(self, coinc_file, bank_file, sngl_files=None, n_loudest=None,
                      group='foreground'):
         self.coinc_file = FileData(coinc_file, group=group)
+        if 'ifos' in self.coinc_file.h5file.attrs:
+            self.ifos = self.coinc_file.h5file.attrs['ifos'].split(' ')
+        else:
+            self.ifos = [self.coinc_file.h5file.attrs['detector_1'],
+                         self.coinc_file.h5file.attrs['detector_2']]
         self.sngl_files = {}
         if sngl_files is not None:
-            for file in sngl_files:
-                curr_dat = FileData(file)
+            if len(sngl_files) < len(self.ifos):
+                raise RuntimeError("Not enough single-detector trigger files "
+                                   "given the IFOs involved in the statmap "
+                                   "file.")
+
+            for sngl_file in sngl_files:
+                curr_dat = FileData(sngl_file)
                 curr_ifo = curr_dat.group_key
                 self.sngl_files[curr_ifo] = curr_dat
+
+            if not sorted(self.sngl_files.keys()) == sorted(self.ifos):
+                logging.warning("WARNING: Statmap file IFOs do not match "
+                                "single-detector trigger files.")
```

Comment:
Why is this not an error?

Reply:
I think that we do need to check if all values of self.ifos are in the sngl_files instead of the check above (e.g. if I give LV triggers with a HL statmap file).

Reply:
By not making it an error it makes it easier to run in the workflow - it means we can just pass a list of all single-detector trigger files without worrying whether the IFO is in the combination or not

I was considering this in the same way that we pass all PTA histogram files to coinc_findtrigs but only use the one which is actually relevant

Reply:
That's a good point - I'll change the len check to be `all([ifo in self.sngl_files.keys() for ifo in self.ifos])` or something like that